vscode/microsoft-kiota/src/steps.ts

```diff
@@ -244,56 +288,62 @@ export async function generateSteps(existingConfiguration: Partial<GenerateState
+                title: `${l10n.t('Create a new API client')} - ${l10n.t('language')}`,
+                step: step++,
+                totalSteps: totalSteps,
+                placeholder: l10n.t('Pick a language'),
+                items,
+                activeItem: typeof state.language === 'string' ? items.find(x => x.languageName === state.language) : undefined,
+                shouldResume: shouldResume
+            });
+            state.language = pick.label.split('-')[0].trim();
+        }
+	}
+    async function inputPluginName(input:MultiStepInput, state: Partial<GenerateState>) {
+        if (!state.pluginName){
+            state.pluginName = await input.showInputBox({
+                title: `${l10n.t('Create a new plugin')} - ${l10n.t('plugin name')}`,
+                step: step++,
+                totalSteps: 3,
+                value: state.pluginName ?? '',
+                placeholder: 'MyPlugin',
+                prompt: l10n.t('Choose a name for the plugin'),
+                validate: validateIsNotEmpty,
+                shouldResume: shouldResume
+            });
+        }
         updateWorkspaceFolder(state.pluginName);
-        return (input: MultiStepInput) => inputPluginType(input, state);
-    }
-    async function inputPluginType(input: MultiStepInput, state: Partial<GenerateState>) {
-        const items = ['API Plugin', 'Open AI'].map(x => ({ label: x }) as QuickPickItem);
-        const pluginTypes = await input.showQuickPick({
-            title: l10n.t('Choose a plugin type'),
-            step: step++,
-            totalSteps: 3,
-            placeholder: l10n.t('Select an option'),
-            items: items,
-            validate: validateIsNotEmpty,
-            shouldResume: shouldResume
-        });
-        pluginTypes.label === 'API Plugin' ? state.pluginTypes = ['ApiPlugin'] : state.pluginTypes = ['OpenAI'];
-        return (input: MultiStepInput) => inputPluginOutputPath(input, state);
-    }
-    async function inputPluginOutputPath(input: MultiStepInput, state: Partial<GenerateState>) {
-        while (true) {
+        return (input: MultiStepInput) => inputPluginType(input, state);      
+    }    
+        async function inputPluginType(input: MultiStepInput, state: Partial<GenerateState>) {
+            if (!state.pluginTypes) {
+                const items = ['API Plugin','Open AI'].map(x => ({ label: x})as QuickPickItem);
+                const pluginTypes = await input.showQuickPick({
+                    title: l10n.t('Choose a plugin type'),
+                    step: step++,
+                    totalSteps: 3,
+                    placeholder: l10n.t('Select an option'),
+                    items: items,
+                    validate: validateIsNotEmpty,
+                    shouldResume: shouldResume
+                });
+                pluginTypes.label === 'API Plugin' ? state.pluginTypes = ['ApiPlugin'] : state.pluginTypes = ['OpenAI'];
```

Comment:
Enum again

Reply:
No, this can not be made an enum. It's a simple if else statement better executed as a conditional operator.

Reply:
I mean an enum defining the constants for the plugin type